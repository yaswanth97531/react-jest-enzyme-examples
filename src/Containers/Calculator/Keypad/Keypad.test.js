import Keypad from './Keypad';

let wrapper;

beforeEach(() => {
  wrapper = mount( 
    <Keypad
      callOperator={jest.fn()}
      numbers={[]}
      operators={[]}
      setOperator={jest.fn()}
      updateDisplay={jest.fn()}
    /> );
});

describe('<Keypad /> rendering', () => {
  it('should render correctly', () => {
    const tree = renderer.create( <Keypad / > ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a <div>', () => {
    expect(wrapper.find('div').length).toEqual(5);
  });

  it('renders the values of numbers', () => {
    wrapper.setProps({ numbers: ['0', '1', '2'] });
    expect(wrapper.find('.numbers-container').text()).toEqual('012');
  });

  it('renders the values of operators', () => {
    wrapper.setProps({operators:['+','-','*','/']});
    expect(wrapper.find('.operators-container').text()).toEqual('+-*/');
  });

  it('should render an instance of the Key component', () => {
    expect(wrapper.find('Key').length).toEqual(1);
  });

  it('should render an instance of the Key component for each index of numbers, operators, and the submit key', () => {
    const numbers = ['0', '1'];
    const operators = ['+', '-'];
    const submit = 1;
    const keyTotal = numbers.length + operators.length + submit;
    wrapper.setProps({ numbers, operators });
    expect(wrapper.find('Key').length).toEqual(keyTotal);
  });
});

describe('<Keypad /> interactions', () => {});

describe('<Keypad /> lifecycle method invocations', () => {});