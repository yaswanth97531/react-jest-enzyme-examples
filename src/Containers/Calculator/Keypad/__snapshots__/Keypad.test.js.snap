// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keypad /> rendering should render correctly 1`] = `
<div
  className="keypad-container"
>
  <div
    className="numbers-container"
  />
  <div
    className="operators-container"
  />
  <div
    className="submit-container"
  >
    <div
      className="key-container submit-key"
      onClick={[Function]}
    >
      <p
        className="key-value"
      >
        =
      </p>
    </div>
  </div>
</div>
`;
